### ISSUE TITLE
3 アーキ infer scripts の _create_pytorch_backend 重複を統合

### ISSUE THREAD

**Issue by kurorosu** [kurorosu/pochidetection#596]
## Summary

3 アーキテクチャ (RT-DETR / SSDLite / SSD300) の `infer.py` に存在する `_create_pytorch_backend()` が, `model load → device → eval → FP16 → backend return` の処理フローで字面ほぼ同一. 共通ヘルパーへの抽出で重複を解消する.

## Background

各アーキで `_create_pytorch_backend()` が独立して実装されており, モデルクラスと backend クラスのみが異なる. 動作は同じため model 構築だけ factory として切り出せば共通化できる.

## Current Implementation

- [pochidetection/scripts/rtdetr/infer.py:156-186](pochidetection/scripts/rtdetr/infer.py#L156-L186)
- [pochidetection/scripts/ssdlite/infer.py:63-88](pochidetection/scripts/ssdlite/infer.py#L63-L88)
- [pochidetection/scripts/ssd300/infer.py:68-93](pochidetection/scripts/ssd300/infer.py#L68-L93)

各ファイルで以下のフローを個別に実装:

```python
def _create_pytorch_backend(...) -> ...Backend:
    model = <ArchModel>(...)
    model.to(device)
    model.eval()
    if use_fp16:
        model = model.half()
    return <ArchBackend>(model, ...)
```

## Proposed Changes

`pochidetection/scripts/_common/` または既存の共通レイヤに `build_pytorch_backend(model_factory, backend_cls, device, use_fp16)` を実装し, 各アーキの `_create_pytorch_backend()` をこのヘルパー呼び出しに置き換える.

## Acceptance Criteria

- 3 アーキの `_create_pytorch_backend()` が共通ヘルパー呼び出しに統一される
- 各アーキ固有の差分 (モデル/バックエンドクラス) が引数で表現される
- 既存テストが pass する

## Labels

`refactor`

## Branch

`refactor/`
